src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -370,6 +370,110 @@ public void It_includes_internals_visible_to()
             AssemblyInfo.Get(assemblyPath)["InternalsVisibleToAttribute"].Should().Be("Tests");
         }
 
+        [Fact]
+        public void Requires_preview_features_enabled()
+        {
+            TestPreviewFeatures(true, true);
+        }
+
+        [Fact]
+        public void Requires_preview_features_disabled()
+        {
+            TestPreviewFeatures(true, false);
+        }
+
+        [Fact]
+        public void Requires_preview_features_not_specified()
+        {
+            TestPreviewFeatures(false, false);
+        }
+
+        private void TestPreviewFeatures(bool addPreviewFeatureProperty, bool enablePreviewFeatures)
+        {
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld", identifier: $"{addPreviewFeatureProperty}_{enablePreviewFeatures}")
+                .WithSource()
+                .WithTargetFramework("net6.0")
+                .WithProjectChanges((path, project) =>
+                {
+                    var ns = project.Root.Name.Namespace;
+
+                    if (addPreviewFeatureProperty)
+                    {
+                        project.Root.Add(
+                            new XElement(ns + "PropertyGroup",
+                                new XElement(ns + "EnablePreviewFeatures", $"{enablePreviewFeatures}")));
+                    }
+                });
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand.Execute().Should().Pass();
+
+            var assemblyPath = Path.Combine(buildCommand.GetOutputDirectory("net6.0").FullName, "HelloWorld.dll");
+
+            var parameterlessAttributes = AssemblyInfo.GetParameterlessAttributes(assemblyPath);
+            bool contains = false;
+            foreach (var attribute in parameterlessAttributes)
+            {
+                if (attribute.Equals("RequiresPreviewFeaturesAttribute", System.StringComparison.Ordinal))
+                {
+                    contains = true;
+                    break;
+                }
+            }
+
+            var getValuesCommand = new GetValuesCommand(testAsset, "LangVersion", targetFramework: "net6.0");
+            getValuesCommand.Execute().Should().Pass();
+
+            var values = getValuesCommand.GetValues();
+            var langVersion = values.FirstOrDefault() ?? string.Empty;
+            if (!addPreviewFeatureProperty || !enablePreviewFeatures)
+            {
+                Assert.False(contains);
+                Assert.True(langVersion.Equals(string.Empty, System.StringComparison.Ordinal));
+            }
+            else
+            {
+                Assert.True(contains);
+                Assert.True(langVersion.Equals("Preview", System.StringComparison.Ordinal));
+            }
+        }
+
+        [Fact]
+        public void It_doesnt_includes_requires_preview_features()
```

Comment:
Is this a duplicate of the coverage already there in the previous tests?

Reply:
It was. I didn't know if there was a difference between defining a property as false (but not using it) vs not defining it at all. Looks like there isn't, so I removed this unit test.